webrender/src/device.rs

```diff
@@ -1562,7 +1563,19 @@ impl Device {
             ImageFormat::Invalid | ImageFormat::RGBAF32 => unreachable!(),
         };
 
-        assert!(data.len() as u32 == bpp * width * height);
+        let row_length = match stride {
+            Some(value) => value / bpp,
+            None => width,
+        };
+
+        assert!(data.len() as u32 == bpp * row_length * height);
+
+        match stride {
```

Comment:
The normal way to write this would be:

```
if let Some(stride) = stride {
 ...
}
```

Then you won't need the empty match case.